imag-todo/src/main.rs

```diff
@@ -16,210 +16,139 @@ extern crate libimagtodo;
-            };
-            for task in iter {
-                match task {
-                    Ok(val) => {
-                        //let val = libimagtodo::task::Task::new(fle);
-                        //println!("{:#?}", val.flentry);
-                        let uuid = match val.flentry.get_header().read("todo.uuid") {
-                            Ok(Some(u)) => u,
-                            Ok(None) => continue,
+                    }
+
+                    if *ttask.status() == TaskStatus::Deleted {
+                        match Task::delete_by_uuid(rt.store(), *ttask.uuid()) {
+                            Ok(_) => println!("Deleted task {}", *ttask.uuid()),
                             Err(e) => {
-                                error!("{}", e);
-                                continue;
+                                trace_error(&e);
+                                exit(1);
                             }
-                        };
-                        if verbose {
-                            args.clear();
-                            args.push(format!("uuid:{}", uuid));
-                            args.push(format!("{}", "information"));
-                            let tw_process = Command::new("task").stdin(Stdio::null()).args(&args).spawn()
-                                .unwrap_or_else(|e| {
-                                    error!("{}", e);
-                                    panic!("failed");
-                                });
-                            let output = tw_process.wait_with_output().unwrap_or_else(|e| {
-                                panic!("failed to unwrap output: {}", e);
-                            });
-                            let outstring = String::from_utf8(output.stdout).unwrap_or_else(|e| {
-                                panic!("failed to ececute: {}", e);
-                            });
-                            println!("{}", outstring);
-                        }
-                        else {
-                            println!("{}", match uuid {
-                                toml::Value::String(s) => s,
-                                _ => {
-                                    error!("Unexpected type for todo.uuid: {}", uuid);
-                                    continue;
-                                },
-                            });
                         }
                     }
+                } // end if c % 2
+            },
+            Err(e) => {
+                trace_error(&e);
+                exit(1);
+            },
+        }
+    } else {
+        // Should not be possible, as one argument is required via
+        // ArgGroup
+        unreachable!();
+    }
+}
+
+fn list(rt: &Runtime) {
+    let subcmd  = rt.cli().subcommand_matches("list").unwrap();
+    let verbose = subcmd.is_present("verbose");
+
+    let res = Task::all(rt.store())
+        .map(|iter| {
+            let uuids : Vec<_> = iter.filter_map(|t| match t {
```

Comment:
Your variable naming is (as always) very minimalistic. How about a comment like 'Get the UUID of all Tasks stored in the imag-store'
Missing comments are nearly the only issues I discover right now, so basically most of your code looks fine to me *thumbs up*. But I'm not finished yet.